code/GridFieldOrderableRows.php

```diff
@@ -505,11 +508,15 @@ protected function reorderItems($list, array $values, array $sortedIDs) {
 			// similar to the SiteTree where you change the position, and then
 			// you go into the record and publish it.
 			foreach($sortedIDs as $sortValue => $id) {
-				if($map[$id] != $sortValue) {
-					$record = $class::get()->byID($id);
+				$record = $class::get()->byID($id);
+				if ($map[$id] != $sortValue) {
 					$record->$sortField = $sortValue;
 					$record->write();
 				}
+				// Update stage to live only for published records
+				if ($record->isPublished()) {
```

Comment:
This is only relevant if there's a write, so you can move it into the condition. Also this is probably where you want to use `$isVersioned` you prepared earlier.